### PR TITLE
Correct message history order in streamed structured response with tool calls

### LINKED ISSUE #234: Message history order is incorrect for streamed structure responses with tool calls
The order of messages from result.all_messages() for streamed structured responses with tool calls is incorrect. The `ToolReturn` message comes before the `ModelStructureResponse` which contains the actual tool call. Tests currently pass because they assert the incorrect order:
https://github.com/pydantic/pydantic-ai/blob/84c1190880219595903df2cea96e5e7146bd715b/tests/test_streaming.py#L225-L237
The expected behaviour is that this test would fail, and the `ModelStructureResponse` message precedes the `ToolReturn` message.

### PR DESCRIPTION
**Pull request by siavashg**
Streamed responses with tool calls would previously order the `ToolReturn` message before the `ModelStructuredResponse` (which invoked the tool call in the first place) in the message history from agent.all_messages(). This corrects the message order.

Fixes #234

### CODE REVIEW

File: tests/test_streaming.py [hunk 1]
```diff
@@ -206,11 +229,6 @@ async def ret_a(x: str) -> str:
                     timestamp=IsNow(tz=timezone.utc),
                 ),
                 ToolReturn(tool_name='ret_a', content='hello world', timestamp=IsNow(tz=timezone.utc)),
-                ToolReturn(
-                    tool_name='final_result',
-                    content='Final result processed.',
-                    timestamp=IsNow(tz=timezone.utc),
-                ),
```
Comment (sydney-runkle) [inline tests/test_streaming.py:213]: Confused why this has been removed...

**Comment by sydney-runkle** [comment]
I'm guessing this might get fixed in https://github.com/pydantic/pydantic-ai/pull/259

File: tests/test_streaming.py (hunk 1 above)
Comment (siavashg) [reply tests/test_streaming.py:213]: The tool call to `final_result` has not been made here yet.
That tool call, and the subsequent ToolReturn for `final_result` (in the right oder now) happens: https://github.com/pydantic/pydantic-ai/pull/233/files/4fbf8b409f425c9eec7b60b9100aec019eb4eeda#diff-d960bb61fde148c9fabda1375f3cb82be40ccc75ac1260cb4353d26cfcedb5d1R243

**Comment by samuelcolvin** [comment]
Principle here looks good, but there's lots of conflicts and I want to change the logic a little, so I'll create a new PR to fix this.

**Comment by sydney-runkle** [comment]
Closing in favor of #274

**Pull request closed without merging**